ivy/__init__.py

```diff
@@ -59,6 +63,22 @@ def __new__(cls, dtype_str):
         return str.__new__(cls, dtype_str)
 
 
+class Shape(tuple):
+    def __new__(cls, shape_tup):
+        assert isinstance(
+            shape_tup, (int, list, tuple, ivy.current_backend().NativeShape)
+        )
+        if isinstance(shape_tup, int):
+            shape_tup = tuple([shape_tup])
```

Comment:
small detail, but there is no need to cast to `list` first. You can just do `shape_tup = (shape_tup,)`